htmd/parameterization/ff.py

```diff
@@ -688,6 +709,17 @@ def __init__(self, mol, prepi, frcmod):
             self.element_by_type[ff[2]] = self._guessElement(ff[1])
             ctr += 1
 
+        # Read improper section
+        with open(prepi) as file:
+            text = file.read()
+        impropers = re.search('^IMPROPER\n(.+)\n\n', text, re.MULTILINE | re.DOTALL)  # extract improper section
+        if impropers:
+            impropers = impropers.group(1).split('\n')  # array of improper lines
+            impropers = [improper.split() for improper in impropers]  # impropers by names
+            for improper in impropers:
+                improper_indices = [self.index_by_name[name.upper()] for name in improper]  # conv atom name to indices
+                self.impropers.append(improper_indices)
+
         f = open(frcmod, "r")
```

Comment:
I agree Stefan, but this is a sort of reader of `FFMolecule`, which sub-classes `Molecule`. Unless we want to have `Molecule` reading force-field files, but that would move functionality of  `FFMolecule` to `Molecule`, which may be what we want, I don't know.
For now, this does not prevent the fix. I'm merging it.